Makefile

```diff
@@ -1,59 +1,59 @@
-sw_objects := ${sensirion_common_dir}/sw_i2c/sensirion_sw_i2c.o ${sensirion_common_dir}/sw_i2c/sensirion_sw_i2c_implementation.o
-hw_objects := ${sensirion_common_dir}/hw_i2c/sensirion_hw_i2c_implementation.o
-all_objects := ${common_objects} ${hw_objects} ${sw_objects} shtc1.o sht3x.o git_version.c
+all: sht3x shtc1
 
-binaries := example_usage_shtc1_sw example_usage_shtc1_hw \
-            example_usage_sht3x_sw example_usage_sht3x_hw
+sht3x/sht3x.o: sht-common/sht_common.h sht3x/sht3x.c
+	cd sht3x && make
 
-.PHONY: FORCE release
+shtc1/shtc1.o: sht-common/sht_common.h shtc1/shtc1.c
+	cd shtc1 && make
 
-git_version.c: FORCE
+sht3x: sht-common/git_version.c sht3x/sht3x.o
+shtc1: sht-common/git_version.c shtc1/shtc1.o
+
+
+sht-common/git_version.c: FORCE
 	git describe --always --dirty | \
 		awk 'BEGIN \
 		{print "/* THIS FILE IS AUTOGENERATED */"} \
 		{print "#include \"git_version.h\""} \
 		{print "const char * SHT_DRV_VERSION_STR = \"" $$0"\";"} \
-		END {}' > git_version.c
-
-all: ${binaries} release
-
-${sensirion_common_dir}/sw_i2c/sensirion_sw_i2c.o: ${sensirion_common_dir}/sw_i2c/sensirion_sw_i2c.c sht.h \
-	${sensirion_common_dir}/sensirion_i2c.h \
-	${sensirion_common_dir}/sw_i2c/sensirion_sw_i2c_gpio.h
-
-shtc1.o: git_version.c shtc1.c sht.h ${sensirion_common_dir}/sensirion_i2c.h
-
-sht3x.o: git_version.c sht3x.c sht.h ${sensirion_common_dir}/sensirion_i2c.h
-
-example_usage.o: example_usage.c sht.h
-
-example_usage_shtc1_sw: ${common_objects} ${sw_objects} shtc1.o
-	$(CC) $(LDFLAGS) -o $@ $^ $(LOADLIBES) $(LDLIBS)
-
-example_usage_sht3x_sw: ${common_objects} ${sw_objects} sht3x.o
-	$(CC) $(LDFLAGS) -o $@ $^ $(LOADLIBES) $(LDLIBS)
-
-example_usage_shtc1_hw: ${common_objects} ${hw_objects} shtc1.o
-	$(CC) $(LDFLAGS) -o $@ $^ $(LOADLIBES) $(LDLIBS)
-
-example_usage_sht3x_hw: ${common_objects} ${hw_objects} sht3x.o
-	$(CC) $(LDFLAGS) -o $@ $^ $(LOADLIBES) $(LDLIBS)
-
-release: git_version.c
-	./release.py && \
-	for directory in release/*; do \
-		echo $${directory}; \
-		cd $${directory}; \
-		export subdir="$$(basename $${directory})"; \
-		export drv=$${subdir%_*_i2c}; \
-		$(CC) $(LDFLAGS) -o example_usage_$${subdir} *.c; \
-		$(RM) example_usage_$${subdir}; \
-		cd -; \
-	done
+		END {}' > $@
+
+
+release/sht3x: sht-common/git_version.c
+	export driver=sht3x && \
+	export tag="$$(git describe --always --dirty)" && \
+	export pkgname="$${driver}-$${tag}" && \
+	export pkgdir="release/$${pkgname}" && \
+	rm -rf "$${pkgdir}" && mkdir -p "$${pkgdir}" && \
+	cp -r embedded-common/* "$${pkgdir}" && \
+	cp -r sht-common/* "$${pkgdir}" && \
+	cp -r $${driver}/* "$${pkgdir}" && \
+	perl -pi -e 's/^sensirion_common_dir :=.*$$/sensirion_common_dir := ./' "$${pkgdir}/Makefile" && \
+	perl -pi -e 's/^sht_common_dir :=.*$$/sht_common_dir := ./' "$${pkgdir}/Makefile" && \
+	cd "$${pkgdir}" && make && make clean && cd - && \
+	cd release && zip -r "$${pkgname}.zip" "$${pkgname}" && cd - && \
+	ln -sf $${pkgname} $@
```

Comment:
Are there any differences between the two release steps beside the pkgname? Otherwise we could extract this to a separate script which takes a parameter.

Reply:
As discussed offline, we can leave it redundant for now and then see if it makes sense to squeeze it into a variable later.